collection-scripts/gather_monitoring

```diff
@@ -1,29 +1,208 @@
+# - GAHTER_MONIT_QUERY_STEP: metric resolution to get from Prometheus. Default: "1m".
+# - GATHER_MONIT_CUSTOM_METRICS: list of additional metrics name to be collected. Default: Undefined
+# - GATHER_MONIT_DISCOVERY_PREFIXES: discovery metrics by prefixes. Default: ''
+# - GAHTER_MONIT_TOKEN: optional service account token to gather Prometheus and Grafana API. Default: Undefined
+#
+# To create the ConfigMap:
+# $ echo -e 'GAHTER_MONIT_START_DATE="15 days ago"\nGAHTER_MONIT_QUERY_STEP="5m"' > env
+# $ echo -e 'GATHER_MONIT_CUSTOM_METRICS="up"' > env
+# $ echo -e 'GATHER_MONIT_DISCOVERY_PREFIXES="apiserver_"' > env
+# $ oc create configmap must-gather-env -n openshift-monitoring --from-file=env=env
+#
+# References:
+# - Prometheus API: https://prometheus.io/docs/prometheus/latest/querying/api/
+
 #Safeguards
-set -o nounset
-set -o errexit
 set -o pipefail
 
-BASE_COLLECTION_PATH="../must-gather"
-MONITORING_PATH="${BASE_COLLECTION_PATH}/monitoring"
+err_report() {
+    echo "ERROR: on line $1 from ${0}"
+}
+trap 'err_report $LINENO' ERR
+
+declare -r GAHTER_MONIT_START_DATE_DEFAULT="7 days ago"
+declare -r GAHTER_MONIT_END_DATE_DEFAULT="now"
+declare -r GAHTER_MONIT_QUERY_STEP_DEFAULT="1m"
+
+declare -r MONIT_NS="openshift-monitoring"
+declare -r MONIT_ENV="must-gather-env"
+
+declare -r CURL_CMD=(curl --fail -sk)
+
+declare -r BASE_COLLECTION_PATH="/must-gather"
+declare -r MONITORING_PATH="${BASE_COLLECTION_PATH}/monitoring"
+declare -r PROM_PATH="${MONITORING_PATH}/prometheus"
+declare -r CA_BUNDLE="${MONITORING_PATH}/ca-bundle.crt"
+
+# Setting: Session token, overwriten by GAHTER_MONIT_TOKEN
+declare -r SS_TOKEN="${GAHTER_MONIT_TOKEN:-$(oc whoami -t)}"
+declare -r SA_TOKEN="$(oc sa get-token default)"
+declare -r PROM_HOST=$(oc -n "${MONIT_NS}" get route prometheus-k8s -o jsonpath='{.spec.host}{"\n"}')
+declare -r PROM_API="/api/v1"
+
+function get_env {
+    oc -n "${MONIT_NS}" extract "cm/${MONIT_ENV}" \
+        --keys=env \
+        --to="${MONITORING_PATH}/" 2>&1 >/dev/null || true
+
+    echo "INFO: Get custom metrics from ConfigMap ${MONIT_ENV} on project ${MONIT_NS}"
+    if [[ -s "${MONITORING_PATH}/env" ]]; then
+        echo "INFO: Loading custom environments variables from ${MONITORING_PATH}/env"
+        source "${MONITORING_PATH}/env"
+    else
+        echo "INFO: Unable to load custom environments from ConfigMap, ignoring."
+    fi
+}
+
+# this is a CA bundle we need to verify the monitoring route,
+# we will write it to disk so we can use it in the flag
+function get_ca_bundle {
+    oc -n openshift-config-managed \
+        get cm default-ingress-cert \
+        -o jsonpath='{.data.ca-bundle\.crt}' > "${CA_BUNDLE}"
+}
+
+function init {
+
+    test -d "${PROM_PATH}" || mkdir -p "${PROM_PATH}"
+
+    get_ca_bundle
+
+    get_env
+
+    # Setting: Timestamp to Prometheus' query range
+    declare -r _START_HUNAN=${GAHTER_MONIT_START_DATE:-${GAHTER_MONIT_START_DATE_DEFAULT}}
```

Comment:
did you mean HUMAN? 

Also if the variable isn't used outside the function, how about making it `local ` ?

Reply:
mtulio~~